### PR TITLE
More const immutable classes

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

**Comment by a14n** [comment]
This addresses https://github.com/flutter/flutter/pull/9247#issue-219798373

>@a14n a lint which would be useful (not for leaving on all the time, but for development) would be one that finds constructors of subclasses of @immutable classes that are not const and whose bodies contain nothing but asserts. :-)

There are still some immutable classes with only asserts in the constructor body but some asserts are not const-compatible so I haven't changed them.

File: packages/flutter/lib/src/widgets/async.dart [hunk 1]
```diff
@@ -309,19 +308,18 @@ class StreamBuilder<T> extends StreamBuilderBase<T, AsyncSnapshot<T>> {
...
-  }
+  }) : assert(builder != null),
+       super(key: key, stream: stream);
 
   /// The build strategy currently used by this builder. Cannot be null.
   final AsyncWidgetBuilder<T> builder;
 
   @override
-  AsyncSnapshot<T> initial() => new AsyncSnapshot<T>.nothing();
+  AsyncSnapshot<T> initial() => new AsyncSnapshot<T>.nothing();// ignore: prefer_const_constructors
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/async.dart]: nit: space before the //

File: packages/flutter/lib/src/widgets/async.dart [hunk 2]
```diff
@@ -412,7 +409,7 @@ class _FutureBuilderState<T> extends State<FutureBuilder<T>> {
   /// calling setState from stale callbacks, e.g. after disposal of this state,
   /// or after widget reconfiguration to a new Future.
   Object _activeCallbackIdentity;
-  AsyncSnapshot<T> _snapshot = new AsyncSnapshot<T>.nothing();
+  AsyncSnapshot<T> _snapshot = new AsyncSnapshot<T>.nothing();// ignore: prefer_const_constructors
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/async.dart]: nit: space before //

**Comment by Hixie** [comment]
nice

**Comment by Hixie** [comment]
LGTM

**Pull request merged**